src/stactools/usda_cdl/download.py

```diff
@@ -0,0 +1,77 @@
+import os
+import pathlib
+from typing import List
+
+import requests
+from tqdm import tqdm
+
+from stactools.usda_cdl.constants import FIRST_AVAILABLE_YEAR, MOST_RECENT_YEAR
+
+
+def download_zips(years: List[int], destination: pathlib.Path) -> List[pathlib.Path]:
+    """Download zipped GeoTiffs from USDA
+
+    Args:
+        years: list of years to download
+        destination: destination directory for downloaded files
+
+    Returns: list of filepaths for downloaded zip files
+    """
+    os.makedirs(str(destination), exist_ok=True)
+    if not years:
+        years = list(range(FIRST_AVAILABLE_YEAR, MOST_RECENT_YEAR + 1))
+    urls = list()
+    for year in years:
+        if year < FIRST_AVAILABLE_YEAR or year > MOST_RECENT_YEAR:
+            raise Exception(f"Unsupported CDL year: {year}")
+        urls.append(
+            "https://www.nass.usda.gov/Research_and_Science/Cropland"
+            f"/Release/datasets/{year}_30m_cdls.zip"
+        )
+        # in 2017 and beyond there is a confidence layer available
+        if year >= 2017:
+            confidence_url = (
+                "https://www.nass.usda.gov/Research_and_Science/Cropland"
+                f"/Release/datasets/{year}_30m_confidence_layer.zip"
+            )
+            # in 2021 they changed the file basename slightly ¯\_(ツ)_/¯
+            if year >= 2021:
+                confidence_url = confidence_url.replace("layer", "Layer").replace(
+                    "confidence", "Confidence"
+                )
+
+            urls.append(confidence_url)
+
+        # each year, a new version of the cumalative (2008-present) "Cultivated"
+        # and "Crop Frequency" layers are generated
```

Comment:
Are the older years' cultivated and crop frequency deleted?

Reply:
USDA only shows the link to the current year's file on the [download page](https://www.nass.usda.gov/Research_and_Science/Cropland/Release/index.php) (right now it's 2008-2022), but it looks like the previous versions are still downloadable.
e.g. https://www.nass.usda.gov/Research_and_Science/Cropland/Release/datasets/Crop_Frequency_2008-2021.zip
https://www.nass.usda.gov/Research_and_Science/Cropland/Release/datasets/Crop_Frequency_2008-2020.zip

Reply:
Ok, in that case we should always (or sometimes) include the cultivated and frequency URLs, not just if `year == MOST_RECENT_YEAR`, correct?

Reply:
Yeah, if you would like to add the old versions of the frequency and cultivated layers to the pile I can see about finding the links. They only display the most recent available file but it looks like the URLs are predictable.